core/src/test/scala/io/chrisdavenport/shellfish/FileOsSpec.scala

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (c) 2024 Typelevel
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package io.chrisdavenport.shellfish
+
+import weaver.SimpleIOSuite
+import weaver.scalacheck.Checkers
+
+import org.scalacheck.Gen
+
+import cats.effect.IO
+
+import fs2.io.file.{Files, CopyFlags}
+
+import syntax.path.*
+import fs2.io.file.CopyFlag
+
+object FileOsSpec extends SimpleIOSuite with Checkers {
+
+  val stringGen: Gen[String] = Gen.asciiStr
+  val contents: String =
+    stringGen.sample.getOrElse("Can't gen the contents 🤷🏻‍♀️")
+
+  test("The API should create and delete a file") {
+    val temp = Files[IO].tempFile
+
+    temp.use { path =>
+      for {
+        _       <- path.write(contents)
+        deleted <- path.deleteIfExists
+      } yield expect(deleted)
+    }
+  }
```

Comment:
can't you use a `forAll` instead of sampling a single string?
```suggestion
  test("The API should create and delete a file") {
   forAll(Gen.asciiStr) { contents => 
    Files[IO].tempFile.use { path =>
      for {
        _       <- path.write(contents)
        deleted <- path.deleteIfExists
      } yield expect(deleted)
    }
  }
```
Also, this specific test doesn't really require a content to be there, just the file, that you can create empty

Reply:
When I used a `forall` method I experienced lots and lots of errors 😭, even when I used the tempFile API inside or outside the `forall` function. I havent tested with the paralellism turned off though